src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/CaseServiceImpl.java

```diff
@@ -43,4 +60,90 @@
 
     return caseData;
   }
+
+  /**
+   * This method contains the business logic for submitting a fulfilment by SMS request.
+   *
+   * @param requestBodyDTO contains the parameters from the originating http POST request.
+   * @throws CTPException if the specified case cannot be found, or if no matching product is found.
+   */
+  @Override
+  public void fulfilmentRequestBySMS(SMSFulfilmentRequestDTO requestBodyDTO) throws CTPException {
+    UUID caseId = requestBodyDTO.getCaseId();
+
+    // Read case from firestore
+    Optional<CollectionCase> caseDetails = dataRepo.readCollectionCase(caseId.toString());
+    if (caseDetails.isEmpty()) {
+      String errorMessage = "Case not found: " + caseId;
+      log.info(errorMessage);
+      throw new CTPException(Fault.RESOURCE_NOT_FOUND, errorMessage);
+    }
+
+    // Attempt to find the requested product
+    Product product =
+        findProduct(caseDetails.get(), requestBodyDTO.getFulfilmentCode(), DeliveryChannel.SMS);
+    if (product == null) {
+      log.info("fulfilmentRequestBySMS can't find compatible product");
+      throw new CTPException(Fault.BAD_REQUEST, "Compatible product cannot be found");
+    }
+
+    // Build and send a fulfilment request event
+    FulfilmentRequest fulfilmentRequestedPayload =
+        createFulfilmentRequestPayload(product, requestBodyDTO.getTelNo(), caseDetails.get());
+    publisher.sendEvent(
+        EventType.FULFILMENT_REQUESTED,
+        Source.RESPONDENT_HOME,
+        Channel.RH,
+        fulfilmentRequestedPayload);
+  }
+
+  // Search the ProductReference for the specified product
+  private Product findProduct(
+      CollectionCase caseDetails, String fulfilmentCode, Product.DeliveryChannel deliveryChannel)
+      throws CTPException {
+
+    Region region = Region.valueOf(caseDetails.getAddress().getRegion());
+    CaseType caseType = CaseType.valueOf(caseDetails.getAddress().getAddressType());
+
+    log.with("region", region)
+        .with("caseType", caseType)
+        .with("deliveryChannel", deliveryChannel)
+        .with("fulfilmentCode", fulfilmentCode)
+        .debug("Attempting to find product.");
+
+    // Build search criteria base on the cases details and the requested fulfilmentCode
+    Product searchCriteria = new Product();
+    searchCriteria.setRequestChannels(Arrays.asList(Product.RequestChannel.RH));
+    searchCriteria.setRegions(Arrays.asList(region));
+    searchCriteria.setCaseType(caseType);
```

Comment:
Don't use case type in product search, as that will limit products to only HH, and we want to support HI subsequently

Reply:
I've removed case type from the product search.